pyiron_base/project/project_browser.py

```diff
@@ -0,0 +1,342 @@
+
+    @property
+    def data(self):
+        return self._data
+
+    def _update_project_worker(self, rel_path):
+        try:
+            new_project = self.project[rel_path]
+            # Check if the new_project implements list_nodes()
+            new_project.list_nodes()
+        except (ValueError, AttributeError):
+            self.path_string_box = self.path_string_box.__class__(description="(rel) Path", value='')
+            with self.output:
+                print("No valid path")
+            return
+        else:
+            if new_project is not None:
+                self.project = new_project
+
+    def _update_project(self, path):
+        if isinstance(path, str):
+            rel_path = os.path.relpath(path, self.path)
+            if rel_path == '.':
+                return
+            self._update_project_worker(rel_path)
+        else:
+            self.project = path
+        self.refresh()
+
+    def _update_pathbox(self, box):
+        path_color = '#DDDDAA'
+        home_color = '#999999'
+
+        def on_click(b):
+            self._busy_check()
+            self._update_project(b.path)
+            self._busy_check(False)
+
+        buttons = []
+        tmppath = os.path.abspath(self.path)
+        if tmppath[-1] == '/':
+            tmppath = tmppath[:-1]
+        len_root_path = len(os.path.split(self._project_root_path[:-1])[0])
+        tmppath_old = tmppath + '/'
+        while tmppath != tmppath_old:
+            tmppath_old = tmppath
+            [tmppath, curentdir] = os.path.split(tmppath)
+            button = widgets.Button(description=curentdir + '/',
+                                    tooltip=curentdir,
+                                    layout=widgets.Layout(width='auto'))
+            button.style.button_color = path_color
+            button.path = tmppath_old
+            button.on_click(on_click)
+            if self.fix_path or len(tmppath) < len_root_path - 1:
+                button.disabled = True
+                if self._hide_path:
+                    button.layout.display = 'none'
+            buttons.append(button)
+        button = widgets.Button(icon="fa-home",
+                                tooltip=self._initial_project_path,
+                                layout=widgets.Layout(width='auto'))
+        button.style.button_color = home_color
+        button.path = self._initial_project
+        if self.fix_path:
+            button.disabled = True
+        button.on_click(on_click)
+        buttons.append(button)
+        buttons.reverse()
+        box.children = tuple(buttons)
+
+    def _on_click_file(self, filename):
+        filepath = os.path.join(self.path, filename)
+        self.output.clear_output(True)
+        try:
+            data = self.project[filename]
+        except(KeyError):
+            data = None
+        with self.output:
+            if data is not None and str(type(data)) == "<class 'boto3.resources.factory.s3.Object'>":
```

Comment:
This logic should be in a separate class and should not be hidden in the GUI logic

Reply:
I will throw it out and try to come up with a nice self-representation of s3 objects. However, this probably requires a wrapped s3-object and I was not sure if this is a good idea.